test/replaytests/replay_test.go

```diff
@@ -98,3 +98,14 @@ func TestGreetingsWorkflow2(t *testing.T) {
 	err := replayer.ReplayWorkflowHistoryFromJSONFile(zaptest.NewLogger(t), "greetings.json")
 	require.Error(t, err)
 }
+
+// this test registers Cherry Activity as the activity but executes a random Activity in the workflow. Ideally replayer doesn't concerns itself with the change in the
+// activity content until it matches the expected output type.History has recorded the output of banana activity instead.
+func TestExclusiveChoiceWorkflow(t *testing.T) {
+	replayer := worker.NewWorkflowReplayer()
+
+	replayer.RegisterWorkflowWithOptions(exclusiveChoiceWorkflow, workflow.RegisterOptions{Name: "choice"})
+	activity.RegisterWithOptions(orderCherryActivity, activity.RegisterOptions{Name: "testactivity", DisableAlreadyRegisteredCheck: true})
```

Comment:
not sure what the goal is with this registration / comment on the test tbh.
registering cherry as an unused activity / activity name should be fine, since it executes `banana`.  it's the same as "there is another activity registered", which is true for all of these tests since there are like 10 activities registered.

Reply:
the order selection change also makes no difference. I think I will add more details in this test case.

When Registered Activity was Cherry but order selection was Banana the test passes.

Reply:
if you mean the order of the cases in the `switch` statement, that's expected - they're exclusive, so order doesn't matter.

which activity is registered also shouldn't matter... normally, as long as the executed one is also registered.

---

Surprisingly the activity is scheduled:
https://github.com/uber-go/cadence-client/pull/1238/files#diff-ef2faa6c08bffac1dce928695bd84dabd73d4f3b8ae614c0be7cfa9d69b0a637R155-R164

Even though there's no blocking call, so it would never be started:
https://github.com/uber-go/cadence-client/pull/1238/files#diff-0e44a72223317107830eacc47b55840ac19e84f35d710475bab01add8e3a6e05R67-R81

so it kinda feels like:
1. the workflow is just irrational, it's not waiting for the activity so it doesn't notice the "called an unregistered activity" case.
    - looks like the sample code is the same: https://github.com/uber-common/cadence-samples/blob/ce920bd6e917aec5fab3c66057a37a51b23c51ef/cmd/samples/recipes/choice/exclusive_choice_workflow.go#L47
    - tbh this probably is worth a test.  this one could work, but a dramatically simpler one would be possible too.
2. so this is currently falling into the "both finished, so it didn't notice the history changed" bug, though for a possibly-different reason (didn't wait for the activity).
    - I'm not sure if this is possible to test separately from 1... but a test does seem useful, yea.  it's sufficiently different from some of the other causes.

Reply:
so I guess: clarify the goal of the commit and test, and we can include it?  it's showing misbehavior, but the reason doesn't match anything that's written down.

Reply:
I racked my brain to figure out how to represent it best but couldn't come up with anything better than simply registering a different activity within the test case. 

currently it looks like the replayer is not bothered by anything that happens inside the switch case.